Lib.AspNetCore.ServerSentEvents/ServerSentEventsMiddleware.cs

```diff
@@ -76,6 +76,13 @@ public async Task Invoke(HttpContext context, IPolicyEvaluator policyEvaluator)
                 {
                     return;
                 }
+                
+                string lastEventId = context.Request.Headers[Constants.LAST_EVENT_ID_HTTP_HEADER];
```

Comment:
It would be nice to extract this into method and have protocol aspects included, something like this:

```
private async Task<bool> ValidateConnectionAsync(HttpContext context)
{
	string lastEventId = context.Request.Headers[Constants.LAST_EVENT_ID_HTTP_HEADER];
	if (!await _serverSentEventsService.OnValidateConnectionAsync(context.Request, lastEventId))
	{
		context.Response.PreventReconnect();

		return false;
	}

	return true;
}
```

Then this becomes only single `if`

```
if (!await ValidateConnectionAsync(context))
{
     return;
 }
```